app/vlstorage/netinsert/netinsert.go

```diff
@@ -325,11 +334,17 @@ func (s *Storage) sendInsertRequestToAnyNode(pendingData *bytesutil.ByteBuffer)
 
 var errTemporarilyDisabled = fmt.Errorf("writing to the node is temporarily disabled")
 
+// RegisterMetricsWriter registers a writer for exposing storage metrics.
+func (s *Storage) RegisterMetricsWriter(w io.Writer) {
+
+}
+
 type streamRowsTracker struct {
 	mu sync.Mutex
 
-	nodesCount    int64
-	rowsPerStream map[uint64]uint64
+	nodesCount       int64
+	totalStreamCount atomic.Uint64
```

Comment:
It looks like this field isn't needed, since it can be inferred from the `rowsPerStream` field as `len(rowsPerStream)` taken under the `mu` lock. This will remove the unnecessary update of this field from hot path at streamRowsTracker.getNodeIdx() below, and simplify the code a bit.